run.py

```diff
@@ -553,67 +157,213 @@ def main():
 
-        if run_bids:
+            #########################################
+            # Get the output.log files and make the physio objects
+            # Set the output dir
+            physio_output_dir = gear_context.output_dir
 
+            # First let's check the output.  We'll try to make this robust for any measurement:
+            gear_context.log.debug('Globbing Physio')
+            all_physio = glob.glob(op.join(physio_output_dir, '*.log'))
+            info_file = glob.glob(op.join(physio_output_dir, '*Info.log'))
+
+            if not info_file:
+                gear_context.log.warning('No Info file found.  Failed Extraction?')
+                return
+            else:
+                info_file = info_file[0]
+
+            # Handle errors and warnings about missing physio data
+            if len(all_physio) == 0 or len(all_physio) == 1:
+                gear_context.log.warning('No physio signals extracted')
+                return
+
+            ######################################################
+            ####   Main Physio Processing/BIDS creation
+            ######################################################
+            info = phys.phys_info(info_file)
+
+            # Add classification for the info file
             try:
-                bids_o_matic_9000(raw_dicom[0],gear_context)
+                data_classifier(gear_context, info.info_dict, info.logfile)
             except Exception as e:
-                gear_context.log.fatal(e, )
-                gear_context.log.fatal('The CMRR conversion to BIDS failed', )
-
-                os.sys.exit(1)
-           # bids_o_matic_9000(raw_dicom[0],gear_context)
+                # We don't want these exceptions to be fatal
+                gear_context.log.info("error updating metadata in log")
+                gear_context.log.exception(e)
+
+            for physio_file in all_physio:
+
+                # If it's the info file, we just made that object, so skip it.
+                if physio_file == info_file:
+                    continue
+
+                physio = phys.physio(physio_file)
+
+                # Set the info object
+                physio.set_info(info)
+                physio.parent_dicom = raw_dicom[0]
+
+                # Add classification for the physio logs
+                try:
+                    data_classifier(gear_context, physio.physio_dict, physio.logfile)
+                except Exception as e:
+                    # We don't want these exceptions to be fatal
+                    gear_context.log.info("error updating metadata in log")
+                    gear_context.log.exception(e)
+
+                # Set some values that will be set by the config file
+                physio.fill_val = gear_context.config['Fill_Value']
+                physio.interp_method = gear_context.config['Interpolation_Method']
+                physio.tic_fill_strategy = gear_context.config["Missing_Data"]
```

Comment:
Somehow, I feel like these could end up as options in your constructor... Same for `physio.set_info(info)`.

I am a big fan of **Encapsulation**...  Hiding things from the user that they do not need to see... 

But who actually reads gear-scripts besides us???

Reply:
you could pass the `gear_context` to you physio object.... but then that would make your physio object dependent on the gear context.

Reply:
Nailed it, that's what I'm trying to avoid.